### DISCUSSION TITLE
charset in .editorconfig

### DISCUSSION THREAD

**Discussion by Youssef1313** [dotnet/runtime#40672] [General] [answered]
The current charset specified in .editorconfig is utf-8.
Would you consider using utf-8-bom based on https://github.com/dotnet/docs/pull/19794#issuecomment-668619665?

More options in .editorconfig are being discussed in https://github.com/dotnet/sdk/pull/12865. (For example, `trim_trailing_whitespace`).

**Comment by danmoseley** [comment, marked as answer]
In this repo, we try to keep all text files in ASCII format, and use means like `\uXXXX` for higher characters. This seems to work well.

**Reply by Youssef1313** [reply]
@danmosemsft Thanks for clarifying. What about trim trailing whitespace?

**Reply by danmoseley** [reply]
I think that’s likely a good rule to add. How easy would it be to throw up a PR?

BTW perhaps I shouldn’t have marked this answered, not sure of the protocol...

**Reply by Youssef1313** [reply]
@danmosemsft that rule already exist, but was discussed that it can cause problems (trimmimg intentional spaces, in verbatim string or markdown for example).

**Reply by danmoseley** [reply]
Ah too bad then.

**Reply by Youssef1313** [reply]
@danmosemsft Ok I'll open a PR to remove it.

**Reply by stephentoub** [reply]
> trimmimg intentional spaces, in verbatim string or markdown for example

Such invisible code is impossible to maintain. And if it exists in C# files, it's going to fail the build due to analyzers we run.
